code/client/src/components/LocalImport.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react'
+import { ImportOutlined, LoadingOutlined } from '@ant-design/icons'
+import { Button, Upload } from 'antd'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router'
+import message from '../message'
+import storage from '../storage'
+import util from '../util'
+import ONEUtil from '../../../lib/util'
+import { walletActions } from '../state/modules/wallet'
+import Paths from '../constants/paths'
+import {LocalExportMessage} from './LocalExport'
+
+const LocalImport = () => {
+  const dispatch = useDispatch()
+  const history = useHistory()
+  const wallets = useSelector(state => state.wallet.wallets)
+  const [fileUploading, setFileUploading] = useState(false)
+
+  const getDataFromFile = file =>
+    new Promise(resolve => {
+      const reader = new FileReader()
+      reader.addEventListener('load', () => resolve(reader.result))
+      reader.readAsArrayBuffer(file)
+    })
+
+  const handleImport = async info => {
+    if (info.file.status === 'uploading') {
+      setFileUploading(true)
+    }
+
+    if (info.file.status === 'done') {
+      try {
+        const data = await getDataFromFile(info.file.originFileObj)
+        const decoded =  LocalExportMessage.decode(new Uint8Array(data))
+        const wallet = JSON.parse(decoded.wallet)
+        const layers = decoded.layers
+
+        if (!util.isValidWallet(wallet)) { throw new Error('Wallet is invalid') }
+        if (util.isWalletExisted(wallets, wallet)) { throw new Error('Wallet is existed') }
+
+        dispatch(walletActions.updateWallet(wallet))
+        storage.setItem(wallet.root, layers)
+        message.success(`Wallet ${wallet.name} (${wallet.address}) is restored!`)
+        setTimeout(() => history.push(Paths.showAddress(wallet.address)), 1500)
+      } catch (err) {
```

Comment:
We should differentiate errors - e.g. failures of reading the file does not mean the wallet is corrupted. It could be device failure (e.g. network/usb drive) or user action (e.g. cancel).